.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Huawei Inc.
+#
+
+name: release
+on:
+  workflow_dispatch:
+    inputs:
+      ReleaseType:
+        type: choice
+        description: Select the version to released
+        options:
+          - Major Version
+          - Minor Version
+          - Patch Version
+
+env:
+  BOT_USER_NAME: eclipse-xpanse-bot
+  BOT_EMAIL_ID: xpanse-bot@eclipse.org
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  pre-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set current version env variable
+        id: version
+        run: |
+          echo "CurrentVersion=$(sed -n 's/^.*version.*= *"\([a-z0-9.]*\)"/\1/pg' version.go)" >> $GITHUB_OUTPUT
+          if [ "${{github.event.inputs.ReleaseType}}" = "Major Version" ]; then
+            echo "VersionFragment=major" >> $GITHUB_ENV
+          elif [ "${{github.event.inputs.ReleaseType}}" = "Minor Version" ]; then
+            echo "VersionFragment=feature" >> $GITHUB_ENV
+          elif [ "${{github.event.inputs.ReleaseType}}" = "Patch Version" ]; then
+            echo "VersionFragment=bug" >> $GITHUB_ENV
+          else
+            echo "No matching feature type  found"
+          fi
+      - name: Set next development version environment variable
+        id: new_version
+        uses: christian-draeger/increment-semantic-version@1.1.0
+        with:
+          current-version: ${{ steps.version.outputs.CurrentVersion }}
+          version-fragment: ${{ env.VersionFragment }}
+      - name: Update version
+        run: |
+          sed -i 's/\(^.*version.*= *"\)\([a-z0-9.]*\)"/\1${{ steps.new_version.outputs.next-version }}"/g' version.go
+      - uses: EndBug/add-and-commit@v9
+        with:
+          message: Release v${{ steps.new_version.outputs.next-version }}
+          committer_name: ${{ env.BOT_USER_NAME }}
+          committer_email: ${{ env.BOT_EMAIL_ID }}
+      - name: Create a GitHub release
```

Comment:
if the purpose of this action is to have only a tag created, then we can consider any other action which creates only a tag and let GoReleaser to handle releases.